src/main/java/internet/shop/dao/impl/CarDaoImpl.java

```diff
@@ -1,54 +1,180 @@
+                PreparedStatement statementForCarsDrivers = connection
+                        .prepareStatement(queryForCarsDrivers)) {
+            statementForCars.setLong(1, id);
+            statementForCarsDrivers.setLong(1, id);
+            return statementForCars.executeUpdate() > 0
+                    && statementForCarsDrivers.executeUpdate() > 0;
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't delete the data by id: " + id, exception);
+        }
     }
 
     public List<Car> getAllByDriver(Long driverId) {
-        return getAll().stream()
-                .filter(car -> car.getDrivers()
-                        .stream()
-                        .anyMatch(driver -> driver.getId().equals(driverId)))
-                .collect(Collectors.toList());
+        List<Car> carList = new ArrayList<>();
+        String query = "select distinct(c.car_id) from cars_drivers cd"
+                + " inner join cars c on cd.car_id = c.car_id "
+                + " where cd.driver_id = ? and c.car_deleted = false;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                Optional<Car> optionalCar = get(resultSet.getObject(1, Long.class));
+                optionalCar.ifPresent(carList::add);
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't find drivers by id: " + driverId, exception);
+        }
+        return carList;
+    }
+
+    private Car getCarFromDB(ResultSet resultSet) throws SQLException {
+        ManufacturerDao manufacturerDao = new ManufacturerDaoImpl();
+        DriverDao driverDao = new DriverDaoImpl();
+        Car car = new Car();
+
+        if (resultSet.next()) {
+            Optional<Manufacturer> optionalManufacturer = manufacturerDao
+                    .get(resultSet.getObject(2, Long.class));
+            optionalManufacturer.ifPresent(car::setManufacturer);
+            car.setId(resultSet.getObject(1, Long.class));
+            car.setModel(resultSet.getObject(3, String.class));
+        }
+        while (resultSet.next()) {
+            Optional<Driver> optionalDriver = driverDao.get(resultSet.getObject(4, Long.class));
+            optionalDriver.ifPresent(car::setDrivers);
+        }
+        return car;
+    }
+
+    private void createCarsDriversDB(Car car) {
+        String queryForCarsDrivers = "INSERT INTO cars_drivers (driver_id, car_id) values (?, ?);";
+        Iterator<Driver> iterator = car.getDrivers().iterator();
+        while (iterator.hasNext()) {
+            try (Connection connection = ConnectionUtil.getConnection();
```

Comment:
let's open only one connection and execute the statement several times in it

Reply:
done